pipelines/tekton/aiedge-e2e/supported-model-formats.json

```diff
@@ -0,0 +1,30 @@
+[
+    {
+        "model-format": "tensorflow",
+        "model-server": "ovms"
```

Comment:
Seldon also seems to support Tensorflow. Would there ever be a case that a user would want to explicitly use that model server for Tensorflow (I guess there would be a few)? If so, then maybe we shouldn't try to force them to use the OpenVino one, right? Maybe we should just expose the model servers that we support, and make the user choose _that_, rather than telling us about the model and then having us choose?

Reply:
I haven't heard anything from the field, but I think it is going to be very likely that users may want to use a specific one. My guess is that the choose may be because of the varying features each model server.

E.g. if a given model would run much faster on a GPU (vs a CPU), OpenVino would be the natural choice if the user infrastructure has Intel GPUs, while Seldon would be the choice for nVidia GPUs. Then, if the infra is mixed, the user may want to build two images containing the same model but using different model servers. This is just an example which may be not very valid for edge-use cases, assuming GPUs would be rare, but I think it works to show that we may need flexibility here.

Reply:
> Maybe we should just expose the model servers that we support, and make the user choose _that_, rather than telling us about the model and then having us choose?

BTW, this is how the odh-dashboard currently works; i.e. you need to choose which model server you want to use when serving a model. For edge, it would be a pipeline rather than directly a model server, but the UX would be quite similar, I think, if we go this route --- So:

* Non-edge case: User picks a model server to serve the desired model
* Edge use case: User picks the pipeline (which inherently means to pick a server) to package the desired model.

I would find the UX so similar up to the point that some users may not realize it is technically different.

Reply:
@israel-hdez For the edge case I think it will be a little different than what you describe, since (at least currently) there's just one pipeline, and the user needs to choose by entering an appropriate parameter value. Based on the context you've provided here, maybe it would be enough to just have a list instead of the map that I've proposed here so far. E.g. something like:

```js
["openvino", "seldon-sklearn", "seldon-mlflow", "seldon-tensorflow", "seldon-triton", "seldon-xgboost", "seldon-tempo"]
```

This way the UI would be able to display meaningful values, but also be able to select the correct base image / Containerfile for the pipeline run.

In the above example, I assume that there would only be one OpenVino base image for all of the model formats that it supports, and no additional configuration is needed for each model format?